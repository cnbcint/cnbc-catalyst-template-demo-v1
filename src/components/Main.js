import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { getMargin, breakpoints } from "../helpers/Layout";
import { ModalContext } from "../components/Modal";

export const LayoutContext = React.createContext({
  maxWidth: "500px",
  colors: null,
});

const Container = styled.div`
  display: block;
  width: 100%;
  background-color: ${(props) => props.colors.background};
  > * {
    font-size: 1.125rem;
    line-height: 140%;
    color: ${(props) => props.colors.copy};
  }
  hr {
    // border-color: ${(props) => props.primaryColor};
    border-color: ${(props) => props.colors.copy};
    border-style: solid;
    display: inline-block;
    width: 100%;
    // margin-top: ${getMargin(1)};
    margin-bottom: ${getMargin(1)};
    @media (min-width: ${breakpoints.mobile}) {
      // margin-top: ${getMargin(2)};
      margin-bottom: ${getMargin(2)};
    }
  }
  img {
    width: 100%;
    display: block;
    margin-top: ${getMargin(2)};
    margin-bottom: ${getMargin(2)};
  }
  p {
    margin: 1rem auto !important;
  }
  ul {
    list-style: disc !important;
    margin-top: ${getMargin(1)} !important;
    margin-bottom: ${getMargin(1)} !important;
    li {
      list-style: disc !important;
      margin: ${getMargin(1)} !important;
      margin-left: ${getMargin(2)} !important;
    }
  }
`;
export default function Main(props) {
  return (
    <LayoutContext.Provider
      value={{ maxWidth: props.maxWidth, colors: props.colors }}
    >
      <ModalContext.Provider value={{ modals: {} }}>
        <Container
          colors={props.colors}
          primaryColor={props.colors.primary}
          className={props.className}
          style={props.style}
        >
          {props.children}
        </Container>
      </ModalContext.Provider>
    </LayoutContext.Provider>
  );
}
Main.defaultProps = {
  maxWidth: "900px",
  colors: { primary: "black", secondary: "gray", tertiary: "red" },
};
Main.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.element,
  ]).isRequired,
  maxWidth: PropTypes.string,
  colors: PropTypes.shape({
    primary: PropTypes.string,
    secondary: PropTypes.string,
    tertiary: PropTypes.string,
    background: PropTypes.string,
    copy: PropTypes.string,
  }),
  style: PropTypes.object,
};
