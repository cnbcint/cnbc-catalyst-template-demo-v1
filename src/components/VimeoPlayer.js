import React from "react";
import PropTypes from "prop-types";
import useIntersectionObserver from "../hooks/useIntersectionObserver";

/**
 Vime embedd player (iframe format) Allow to embed a vimeo video
*/
export default function VimeoPlayer(props) {
  const styles = {
    padding: "56.25% 0 0 0",
    position: "relative",

    ...props.style,
  };

  const { nodeRef, isIntersecting } = useIntersectionObserver();
  let autoplay = false;
  if (props.autoplayScroll) {
    autoplay = isIntersecting;
  } else if (props.autoplayLoad) {
    autoplay = true;
  }
  return (
    <div style={styles} className={props.className}>
      {props.coverImage && (
        <div
          style={{
            backgroundImage: `url(${props.coverImage})`,
            position: "absolute",
            width: "100%",
            height: "100%",
            top: 0,
            backgroundSize: "cover",
          }}
          className={props.className ? props.className + "__cover" : ""}
        />
      )}
      <iframe
        ref={nodeRef}
        src={`https://player.vimeo.com/video/${props.videoID}?background=${
          props.background ? 1 : 0
        }&autoplay=${autoplay ? 1 : 0}&title=${
          props.showTitle ? 1 : 0
        }&byline=${props.showByLine ? 1 : 0}&portrait=${
          props.showPortrait ? 1 : 0
        }`}
        style={{
          position: "absolute",
          top: 0,
          left: 0,
          width: "100%",
          height: "100%",
        }}
        frameBorder="0"
        allow="autoplay;"
        allowFullScreen
      ></iframe>
      {props.overlay && (
        <div
          style={{
            backgroundColor: props.overlayColor,
            position: "absolute",
            top: 0,
            width: "100%",
            height: "100%",
            left: 0,
          }}
          className={props.className ? props.className + "__overlay" : ""}
        ></div>
      )}
    </div>
  );
}

VimeoPlayer.defaultProps = {
  className: "",
  background: false,
  autoplayLoad: false,
  autoplayScroll: false,
  overlay: false,
  overlayColor: "rgba(0,0,0,.5)",
  showTitle: false,
  showByLine: false,
  showPortrait: false,
  style: {},
};

VimeoPlayer.propTypes = {
  className: PropTypes.string,

  /**
   * Vimeo Video ID
   */
  videoID: PropTypes.string.isRequired,

  /**
   * Use for background video. This will remove the controls and autoplay the video muted
   */
  background: PropTypes.bool,

  /**
   * Autoplay video on load of the page
   */

  autoplayLoad: PropTypes.bool,
  /**
   * Autoplay video on enteriving viewportt
   */

  autoplayScroll: PropTypes.bool,
  /**
   * When using background you can addd an overlay div on top of video and customize color with overlayColor
   */
  overlay: PropTypes.bool,

  /**
   * RGBA css property for overlay color this will enabled to darken the video for backgrounds.
   */
  overlayColor: PropTypes.string,

  /**
   * Displays a cover image before the background video is loaded
   */
  coverImage: PropTypes.string,

  /**
   * Show the video's title.
   */
  showTitle: PropTypes.bool,

  /**
   * Show the author of the video
   */
  showByLine: PropTypes.bool,

  /**
   * Show the author's profile image (portrait)
   */
  showPortrait: PropTypes.bool,

  /**
  React CSS format to override rules has required by design
  */
  style: PropTypes.object,
};
