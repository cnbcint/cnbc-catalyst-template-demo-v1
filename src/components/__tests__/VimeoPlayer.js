import React from "react";
import VimeoPlayer from "../VimeoPlayer";
import Main from "../Main";
import { render } from "@testing-library/react";
import "jest-styled-components";

const Theme = {
  maxWidth: "900px",
  colors: {
    primary: "orange",
    secondary: "green",
    tertiary: "orange",
    background: "white",
    copy: "black",
    links: "blue",
  },
};
const Component = () => (
  <Main {...Theme}>
    <VimeoPlayer
      className="VimeoPlayerTest"
      overlay
      overlayColor="rgba(0, 0, 0, 0.5)"
      videoID="468693933"
      coverImage="http://localhost:8080/images/header-desktop.jpg"
      background
      autoplayLoad
    />
  </Main>
);

beforeAll(() => {
  window.IntersectionObserver = class {
    observe() {
      return null;
    }
    disconnect() {}
  };
});

describe("VimeoPlayer", () => {
  it("Renders correctly", () => {
    const { container } = render(<Component />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Renders a video", () => {
    const { container } = render(<Component />);
    const iframe = container.getElementsByTagName("iframe")[0];
    const src = iframe.getAttribute("src");
    expect(src).toContain("https://player.vimeo.com/video/468693933");
  });

  it("Renders a cover image", () => {
    const { container } = render(<Component />);

    const coverImage = container.getElementsByClassName(
      "VimeoPlayerTest__cover"
    )[0];

    expect(coverImage.style["background-image"]).toBe(
      "url(http://localhost:8080/images/header-desktop.jpg)"
    );
  });

  it("Renders the overlay with the correct color", () => {
    const { container } = render(<Component />);

    const overlay = container.getElementsByClassName(
      "VimeoPlayerTest__overlay"
    )[0];
    expect(overlay.style["background-color"]).toBe("rgba(0, 0, 0, 0.5)");
  });

  it("Sends parameter background", () => {
    const { container } = render(<Component />);
    const iframe = container.getElementsByTagName("iframe")[0];
    const src = iframe.getAttribute("src");
    expect(src).toContain("background=1");
  });

  it("Sends parameter autoplay", () => {
    const { container } = render(<Component />);
    const iframe = container.getElementsByTagName("iframe")[0];
    const src = iframe.getAttribute("src");
    expect(src).toContain("autoplay=1");
  });

  it("Adds classNames ", () => {
    const { container } = render(<Component />);
    const target = container.getElementsByClassName("VimeoPlayerTest")[0];
    expect(target.classList.contains("VimeoPlayerTest")).toBe(true);
  });
});

afterAll(() => {
  window.IntersectionObserver = null;
});
