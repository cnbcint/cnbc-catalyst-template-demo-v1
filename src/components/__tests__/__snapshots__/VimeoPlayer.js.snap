// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VimeoPlayer Renders correctly 1`] = `
.c0 {
  display: block;
  width: 100%;
  background-color: white;
}

.c0 > * {
  font-size: 1.125rem;
  line-height: 140%;
  color: black;
}

.c0 hr {
  border-color: orange;
  border-style: solid;
  display: inline-block;
  width: 100%;
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.c0 img {
  width: 100%;
  display: block;
  margin-top: 2rem;
  margin-bottom: 2rem;
}

.c0 p {
  margin-top: 1rem !important;
  margin-bottom: 1rem !important;
}

.c0 ul {
  list-style: disc !important;
  margin-top: 1rem !important;
  margin-bottom: 1rem !important;
}

.c0 ul li {
  list-style: disc !important;
  margin: 1rem !important;
  margin-left: 2rem !important;
}

@media (min-width:768px) {
  .c0 hr {
    margin-top: 2rem;
    margin-bottom: 2rem;
  }
}

<div
  class="c0"
>
  <div
    class="VimeoPlayerTest"
    style="padding: 56.25% 0px 0px 0px; position: relative;"
  >
    <div
      class="VimeoPlayerTest__cover"
      style="background-image: url(http://localhost:8080/images/header-desktop.jpg); position: absolute; width: 100%; height: 100%; top: 0px; background-size: cover;"
    />
    <iframe
      allow="autoplay;"
      allowfullscreen=""
      frameborder="0"
      src="https://player.vimeo.com/video/468693933?background=1&autoplay=1&title=0&byline=0&portrait=0"
      style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
    />
    <div
      class="VimeoPlayerTest__overlay"
      style="background-color: rgba(0, 0, 0, 0.5); position: absolute; top: 0px; width: 100%; height: 100%; left: 0px;"
    />
  </div>
</div>
`;
